airflow/executors/mesos_executor.py

```diff
@@ -0,0 +1,380 @@
+        )
+
+        master_urls = "https://" + master
+
+        self.client = MesosClient(
+            mesos_urls=master_urls.split(','),
+            frameworkName=framework_name,
+            frameworkId=None,
+        )
+
+        self.client.set_role(framework_role)
+
+        if framework_checkpoint:
+            self.client.set_checkpoint(framework_checkpoint)
+            self.client.set_failover_timeout(framework_failover_timeout)
+
+        if conf.getboolean('mesos', 'AUTHENTICATE'):
+            if not conf.get('mesos', 'DEFAULT_PRINCIPAL'):
+                self.log.error("Expecting authentication principal in the environment")
+                raise AirflowException("mesos.default_principal not provided in authenticated mode")
+            if not conf.get('mesos', 'DEFAULT_SECRET'):
+                self.log.error("Expecting authentication secret in the environment")
+                raise AirflowException("mesos.default_secret not provided in authenticated mode")
+            self.client.principal = conf.get('mesos', 'DEFAULT_PRINCIPAL')
+            self.client.secret = conf.get('mesos', 'DEFAULT_SECRET')
+
+        driver = AirflowMesosScheduler(self, self.task_queue, self.result_queue, task_cpu, task_memory)
+        self.driver = driver
+        self.client.on(MesosClient.SUBSCRIBED, driver.subscribed)
+        self.client.on(MesosClient.UPDATE, driver.status_update)
+        self.client.on(MesosClient.OFFERS, driver.resource_offers)
+        self.mesos_framework = MesosExecutor.MesosFramework(self.client)
+        self.mesos_framework.start()
+
+    def sync(self) -> None:
+        """Updates states of the tasks."""
+        self.log.debug("Update state of tasks")
+        if self.running:
+            self.log.debug('self.running: %s', self.running)
+
+        while not self.result_queue.empty():
+            results = self.result_queue.get()
+            key, state = results
+            if state == "success":
+                self.log.info("tasks successfull %s", key)
+                self.task_queue.task_done()
+            if state == "failed":
+                self.log.info("tasks failed %s", key)
+                self.task_queue.task_done()
+            self.change_state(*results)
+
+    def execute_async(
+        self,
+        key: TaskInstanceKey,
+        command: CommandType,
+        queue: Optional[str] = None,
+        executor_config: Optional[Any] = None,
+    ):
+        """Execute Tasks"""
+        self.log.info('Add task %s with command %s with TaskInstance %s', key, command, executor_config)
+        self.validate_command(command)
+        self.task_queue.put((key, command, executor_config))
+
+    def end(self) -> None:
+        """Called when the executor shuts down"""
+        self.log.info('Shutting down Mesos Executor')
+        # Both queues should be empty...
+        self.task_queue = Queue()
+        self.result_queue = Queue()
+        self.task_queue.join()
+        self.result_queue.join()
```

Comment:
What's the point of joining a freshly created queue? I don't think you wanted to reset these attributes here.

Reply:
correct. I fix it.